modules/nf-core/vizgenpostprocessing/preparesegmentation/main.nf

```diff
@@ -0,0 +1,60 @@
+process VIZGENPOSTPROCESSING_PREPARESEGMENTATION {
+    tag "$meta.id"
+    label 'process_single'
+
+    container 'ghcr.io/wehi-soda-hub/vizgen-postprocessing_container:v0.1.1'
+
+    input:
+    val(meta)
+    path(algorithm_json)
+    path(input_images)
+    val(images_regex)
+    path(um_to_mosaic_file)
+    val(tile_size)
+    val(tile_overlap)
+
+    output:
+    tuple val(meta), path("*.json"), path(input_images), path(algorithm_json), emit: segmentation_files
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    // Exit if running this module with -profile conda / -profile mamba
+    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
+        error "VPT is unavailable via Conda. Please use Docker / Singularity / Apptainer / Podman instead."
+    }
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    vpt --verbose \\
+        prepare-segmentation \\
+        $args \\
+        --segmentation-algorithm $algorithm_json \\
+        --input-images "${input_images}/${images_regex}" \\
+        --input-micron-to-mosaic $um_to_mosaic_file \\
+        --output-path . \\
+        --tile-size $tile_size \\
+        --tile-overlap $tile_overlap
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        vpt: \$( pip show vpt | grep Version | sed -e "s/Version: //g" )
+        vpt-plugin-cellpose2: \$( pip show vpt-plugin-cellpose2 | grep Version | sed -e "s/Version: //g" )
+    END_VERSION
+    """
+
+    stub:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    touch algorithm_specification.json
```

Comment:
Is it the only file produce by the tool ?

Reply:
Yes, it only creates a spec file with metadata for a segmentation run. The next step (`run-segmentation-on-tile`) actually runs the segmentation on a given tile. I plan to contribute this as the next module for this tool.